scripts/library.sh

```diff
@@ -132,30 +132,47 @@ function wait_until_object_does_not_exist() {
 # Parameters: $1 - namespace.
 function wait_until_pods_running() {
   echo -n "Waiting until all pods in namespace $1 are up"
+  local failed_pod=""
   for i in {1..150}; do  # timeout after 5 minutes
     local pods="$(kubectl get pods --no-headers -n $1 2>/dev/null)"
     # All pods must be running
-    local not_running=$(echo "${pods}" | grep -v Running | grep -v Completed | wc -l)
+    local not_running_pods=$(echo "${pods}" | grep -v Running | grep -v Completed)
+    local not_running=$(echo "${not_running_pods}" | wc -l)
     if [[ -n "${pods}" && ${not_running} -eq 0 ]]; then
+      # All Pods are running or completed. Verify the containers on each Pod.
       local all_ready=1
       while read pod ; do
         local status=(`echo -n ${pod} | cut -f2 -d' ' | tr '/' ' '`)
+        local pod_name=$(echo -n ${pod} | cut -f1 -d' ')
         # All containers must be ready
-        [[ -z ${status[0]} ]] && all_ready=0 && break
-        [[ -z ${status[1]} ]] && all_ready=0 && break
-        [[ ${status[0]} -lt 1 ]] && all_ready=0 && break
-        [[ ${status[1]} -lt 1 ]] && all_ready=0 && break
-        [[ ${status[0]} -ne ${status[1]} ]] && all_ready=0 && break
+        [[ -z ${status[0]} ]] && all_ready=0 && failed_pod="${pod_name}" && break
+        [[ -z ${status[1]} ]] && all_ready=0 && failed_pod="${pod_name}" && break
+        [[ ${status[0]} -lt 1 ]] && all_ready=0 && failed_pod="${pod_name}" && break
+        [[ ${status[1]} -lt 1 ]] && all_ready=0 && failed_pod="${pod_name}" && break
+        [[ ${status[0]} -ne ${status[1]} ]] && all_ready=0 && failed_pod="${pod_name}" && break
       done <<< "$(echo "${pods}" | grep -v Completed)"
       if (( all_ready )); then
         echo -e "\nAll pods are up:\n${pods}"
         return 0
       fi
+    elif [[ ${not_running} -ne 0 ]]; then
+      # At least one Pod is not running, just save the first one's name as the failed_pod.
+      while read pod ; do
+        local pod_name=$(echo -n ${pod} | cut -f1 -d' ')
+        failed_pod="${pod_name}"
+        break
+      done <<< "$(echo "${not_running_pods}")"
     fi
     echo -n "."
     sleep 2
   done
   echo -e "\n\nERROR: timeout waiting for pods to come up\n${pods}"
+  if [[ -n "${failed_pod}" ]]; then
+    echo -e "\n\nFailed pod oyaml - ${failed_pod}\n"
```

Comment:
"Failed pod oyaml" sounds like a typo to me. :) Maybe something like "Failed pod (data in YAML format)"?